### PR TITLE
Discord Verification Changes

### PR DESCRIPTION
**Pull request by maxkross**
Here are the changes I made to get Discord Authentication working

### CODE REVIEW

File: constants.py
Comment (EpicWolverine) [inline constants.py:6]: It's a goofy way of doing this but keep the comment for now since the code looks for those exact strings.

File: flask_app.py
Comment (EpicWolverine) [inline flask_app.py]: I'm not sure I've ever seen this code path taken, but it should probably use ERROR_PAGE instead
```python
return build_template_response(None, ERROR_PAGE, template_params)
```

File: flask_app.py
Comment (EpicWolverine) [inline flask_app.py]: Ope this is the Michigan Risk Discord. Not sure what would be better. Maybe we get a dedicated channel on the main Risk Discord. Or better yet, we move this to a constant/config value so each deployment can set their own support channel.

File: flask_app.py
Comment (EpicWolverine) [inline flask_app.py]: Probably best to update names like `authlink` and `make_authorization_url()` to `redditauthlink` and `make_reddit_authorization_url()` now that we have 2.

File: flask_app.py
Comment (EpicWolverine) [inline flask_app.py:280]: Ohh it just tries to auth with Reddit and then Discord every time? I don't like that. Maybe we need to add another cookie that says which service the access_token is associated with.

File: flask_app.py
Comment (EpicWolverine) [inline flask_app.py:295]: Yeah, see above. I think we can be smarter about this.

File: flask_app.py
Comment (EpicWolverine) [inline flask_app.py:314]: This is so similar to `get_token()`, we could probably add a couple arguments and reduce the duplicate code.

File: flask_app.py
Comment (EpicWolverine) [inline flask_app.py]: This shouldn't be necessary if `check_identity_or_auth()` is rewritten, but still probably a good check. It could be shorter though.

```suggestion
    # If reddit or discord fail, then don't return anything
    try:
        return response.json()['name']
    except:
        return None
```

**Review by EpicWolverine — CHANGES_REQUESTED** [review]
This fixes https://github.com/Tapin42/cfbr_orders/issues/61

Also I don't have write access so I can't merge yet anyway

File: flask_app.py
Comment (maxkross) [reply flask_app.py]: Added GOOD_GUYS_DISCORD_LINK in .env

File: flask_app.py
Comment (maxkross) [reply flask_app.py:280]: I don't really know how to detect which button they clicked, could you help me with that piece?

Comment (maxkross) [reply flask_app.py:280]: Oh... unless we put it in the callback route? Maybe I can figure that out.

File: flask_app.py
Comment (maxkross) [reply flask_app.py:314]: I feel like the functions are short enough that this helps with readability. Otherwise you're be shoving params into an argument. I think this justifies the duplicate code but let me know your thoughts.

File: flask_app.py
Comment (maxkross) [reply flask_app.py:295]: Changed this to use new cookie

File: flask_app.py
Comment (EpicWolverine) [inline flask_app.py:147]: Unless I’m missing it, `template_params` doesn’t exist here. You’ll need to pass in a dict with at least `{“error_message” = error}`

File: flask_app.py
Comment (EpicWolverine) [inline flask_app.py]: I think `a` was short for “access token”? Maybe it’s a standard I don’t know about, but it’s not a great name, but not worth changing. This however should have a good name instead of `b`. Perhaps `service` or `auth_site` like you use later. 
Also, putting this name into a constant at the top of the file will help having to change it in multiple spots in the future.

File: flask_app.py
Comment (EpicWolverine) [inline flask_app.py]: You missed this one.

File: flask_app.py
Comment (EpicWolverine) [reply flask_app.py:280]: If I’m reading this right (the formatting is a bit scrunched on my phone), we’re still checking reddit and then failing over to discord. Now that we know which service we’re supposed to be authenticated with from the cookie, we can just only do the appropriate one and immediately redirect to the auth page if the service 401s.

File: flask_app.py
Comment (EpicWolverine) [reply flask_app.py:314]: It’s kinda long by my standards lol, but don’t worry about it. I can refactor it later  and we can discuss since it will be easier than describing what I have in my head.

**Review by EpicWolverine — CHANGES_REQUESTED** [review]

File: flask_app.py
Comment (maxkross) [reply flask_app.py]: If we're giving them real names we're going to give both real names ;)

Comment (EpicWolverine) [reply flask_app.py]: I support this.

File: flask_app.py
Comment (EpicWolverine) [inline flask_app.py:281]: It's good to have an `else:` clause to fall back on and give us a good error in case there's some case where `auth-site` is not set or not one of the supported values, otherwise you're going to get a NameError below that `response` is not defined.

File: flask_app.py
Comment (EpicWolverine) [inline flask_app.py:185]: Making `access-token` and `auth-site` constants would also be wise.

**Review by EpicWolverine — APPROVED** [review]
I think this is good enough to merge or I'll just keep polishing it. The rest of my feedback can be implemented later.

**Pull request merged**
